File: app/src/main/java/com/example/loginsample2/fragments/CuadrosFragment.java
package com.example.loginsample2.fragments;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SearchView;

import com.example.loginsample2.AdapterRecyclerView;
import com.example.loginsample2.FragmentChanger;
import com.example.loginsample2.R;

import java.util.ArrayList;
import java.util.List;

public class CuadrosFragment extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    private RecyclerView recyclerView;
    private AdapterRecyclerView adapterRecyclerView;
    private List<AdapterRecyclerView.Item> items;

    private static FragmentChanger fragmentChanger;

    public CuadrosFragment() {
        // Required empty public constructor
    }

    public static CuadrosFragment newInstance(String a, String b, FragmentChanger fragmentChanger) {
        CuadrosFragment fragment = new CuadrosFragment();
        Bundle args = new Bundle();
        CuadrosFragment.fragmentChanger = fragmentChanger;
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view=inflater.inflate(R.layout.fragment_cuadros, container, false);

        SearchView searchView= view.findViewById(R.id.searchView);
        recyclerView=view.findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        //ITEMS DE EJEMPLO
        items= new ArrayList<>();
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));
        items.add(new AdapterRecyclerView.Item(1,"Carpintero de Nidos","Sala N° X", "Daniel Gallegos Esquivias"));



        adapterRecyclerView=new AdapterRecyclerView(items, new AdapterRecyclerView.OnImageClickListener() {
            @Override
            public void onImageClick(int imageId) {
                fragmentChanger.changeFragment(CuadroDetalleFragment.newInstance("",""));
            }
        });
        recyclerView.setAdapter(adapterRecyclerView);

        return view;
    }
}

File: app/src/main/java/com/quantumsoft/myapplication/fragments/CuadrosFragment.kt
package com.quantumsoft.myapplication.fragments
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.SearchView
import com.quantumsoft.myapplication.AdapterRecyclerView
import com.quantumsoft.myapplication.FragmentChanger
import com.quantumsoft.myapplication.R
import com.quantumsoft.myapplication.model.data.Pintura
import com.quantumsoft.myapplication.viewmodel.MuseoViewModel
import java.util.ArrayList

class CuadrosFragment : Fragment() {

    private var mParam1: String? = null
    private var mParam2: String? = null

    private lateinit var recyclerView: RecyclerView
    private lateinit var adapterRecyclerView: AdapterRecyclerView
    private lateinit var items: MutableList<AdapterRecyclerView.Item>


    companion object {
        lateinit var fragmentChanger: FragmentChanger
        lateinit var museoViewModel: MuseoViewModel

        @JvmStatic
        fun newInstance(a: String, b: String, viewmodel: MuseoViewModel, fragmentChanger: FragmentChanger): CuadrosFragment {
            val fragment = CuadrosFragment()
            CuadrosFragment.fragmentChanger = fragmentChanger
            val args = Bundle()
            fragment.arguments = args
            museoViewModel = viewmodel
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            mParam1 = it.getString(mParam1)
            mParam2 = it.getString(mParam2)
        }





    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_cuadros, container, false)

        val searchView: SearchView = view.findViewById(R.id.searchView)
        recyclerView = view.findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(requireContext())

        // ITEMS DE EJEMPLO
        items = ArrayList()
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))
//        items.add(AdapterRecyclerView.Item(1, "Carpintero de Nidos", "Sala N° X", "Daniel Gallegos Esquivias"))

        museoViewModel.addExampleData()
        adapterRecyclerView = AdapterRecyclerView(items, AdapterRecyclerView.OnImageClickListener { imageId ->
            fragmentChanger.changeFragment(CuadroDetalleFragment.newInstance("", ""))
        })
        recyclerView.adapter = adapterRecyclerView

        museoViewModel.updateListItems()
        museoViewModel.items.observe(viewLifecycleOwner) {
            adapterRecyclerView.updateList(it)
        }

        return view
    }
}
